nrf_security/Kconfig.tls

```diff
@@ -0,0 +1,283 @@
+	  identify DTLS connections across changes in the underlying transport.
+	  The Connection ID extension is still in draft state.
+
+config MBEDTLS_SSL_DTLS_BADMAC_LIMIT
+	bool
+	prompt "SSL - DTLS bad MAC limit"
+	help
+	  Enable support for a limit of records with bad MAC.
+
+endif # MBEDTLS_SSL_PROTO_DTLS
+
+config MBEDTLS_SSL_ALL_ALERT_MESSAGES
+	bool
+	prompt "Enable all SSL alert messages"
+	default y
+	help
+	  This setting enables alert messages beingsentin case of failures.
+	  Disabling this will enure that no information is given about reasons
+	  for failures.
+	  Corresponds to MBEDTLS_SSL_ALL_ALERT_MESSAGES in mbed TLS config file.
+
+config MBEDTLS_SSL_CONTEXT_SERIALIZATION
+	bool
+	prompt "Enable serialization of TLS conftext structures"
+	default y
+	help
+	  This setting enables serialization of TLS structures, throught use of
+	  the mbedtls_ssl_context_save() and mbedtls_ssl_context_load().
+	  Corresponds to MBEDTLS_SSL_CONTEXT_SERIALIZATION in mbed TLS config file.
+
+config MBEDTLS_SSL_DEBUG_ALL
+	bool
+	prompt "Enable debug messages in SSL module for all issues."
+	help
+	  This setting enable the debug messages in SSL module for all issues.
+	  Note: Some debug messages have been enabled to prevent timing attacks.
+	  Corresponds to MBEDTLS_SSL_DEBUG_ALL in mbed TLS config file.
+
+config MBEDTLS_SSL_KEEP_PEER_CERTIFICATE
+	bool
+	prompt "Give acces to peer certificate after handshake"
+	default y
+	help
+	  Provides mbedtls_ssl_get_peer_cert() the ability to give the
+	  peer's certificate after completion of handshake. Disabling this
+	  reduces RAM usage.
+	  Corresponds to MBEDTLS_SSL_KEEP_PEER_CERTIFICATE in mbed TLS config file.
+
+config MBEDTLS_SSL_RENEGOTIATION
+	bool
+	prompt "SSL - Renegotiation"
+	help
+	  Enable support for TLS renegotiation.
+
+config MBEDTLS_SSL_SESSION_TICKETS
+	bool
+	prompt "SSL - RFC 5077 session tickets support"
+	help
+	  Enable support for RFC 5077 session tickets in SSL.
+
+config MBEDTLS_SSL_SERVER_NAME_INDICATION
+	bool
+	prompt "SSL - RFC 6066 SNI"
+	depends on MBEDTLS_X509_CRT_PARSE_C
+	help
+	  Enable support for RFC 6066 server name indication (SNI) in SSL.
+
+config MBEDTLS_SSL_CACHE_C
+	bool
+	prompt "SSL - cache"
+	help
+	  Enable simple SSL cache implementation.
+
+config MBEDTLS_SSL_TICKET_C
+	bool
+	prompt "SSL - tickets"
+	depends on MBEDTLS_CIPHER_C
+	help
+	  Enable an implementation of TLS server-side callbacks for session tickets.
+
+
+config MBEDTLS_SSL_EXPORT_KEYS
+	prompt "Enable support for exporting keys"
+	default y
+	help
+	  Enables suppport for exporting key block and master secret.
+
+
+config MBEDTLS_SSL_OUT_CONTENT_LEN
+	prompt "Max length for TLS outgoing fragments"
+	int
+	range 0 16384
+	default 900 if OPENTHREAD_NRF_SECURITY
+	default 16380
+
+config MBEDTLS_SSL_IN_CONTENT_LEN
+	prompt "Max length for TLS outgoing fragments"
+	int
+	range 0 16384
+	default 900 if OPENTHREAD_NRF_SECURITY
+	default 16380
```

Comment:
This constant (16380) leads to mbedtls not working after this fix https://github.com/nrfconnect/sdk-mbedtls/commit/668b31f210356ef26b1ca26bced775b3d36f63f9

(unless of course server sends less)

Glad this config and wrong constant is removed altogether now.

But with SDK-NRF 2.4.2 it's not working...